tfhe/src/high_level_api/glwe_ct_list.rs

```diff
@@ -0,0 +1,211 @@
+use super::global_state::with_cpu_internal_keys;
+use crate::core_crypto::commons::math::random::{Deserialize, Serialize};
+use crate::high_level_api::integers::{FheIntId, FheUintId};
+use crate::integer::ciphertext::{DataKind, Expandable, GlwePackable};
+use crate::named::Named;
+use crate::shortint::Ciphertext;
+use crate::{FheBool, FheInt, FheUint};
+
+impl<Id: FheUintId> GlwePackable for FheUint<Id> {
+    fn compact_into(self, messages: &mut Vec<Ciphertext>) -> DataKind {
+        self.ciphertext.on_cpu().to_owned().compact_into(messages)
```

Comment:
Since you take the FheUint by `self` you should be able to do `self.ciphertext.into_cpu()` this will avoid the clone done in to_owned